src/trio/_tests/test_dtls.py

```diff
@@ -708,14 +751,15 @@ def route_packet(packet: UDPPacket) -> None:
             print(f"delivering {packet}")
             fn.deliver_packet(packet)
 
-    fn.route_packet = route_packet  # type: ignore[assignment]  # TODO add type annotations for FakeNet
+    fn.route_packet = route_packet  # type: ignore[assignment]  # TODO: add type annotations for FakeNet
 
     # See if we can successfully do a handshake -- some of the volleys will get dropped,
     # and the retransmit logic should detect this and back off the MTU to something
     # smaller until it succeeds.
-    async with dtls_echo_server(ipv6=ipv6) as (_, address):
+    async with dtls_echo_server(ipv6=ipv6, server_ctx=server_ctx) as (_, address):
         with endpoint(ipv6=ipv6) as client_endpoint:
-            client = client_endpoint.connect(address, client_ctx)
+            # TODO: why is making a new client context necessary here?
+            client = client_endpoint.connect(address, client_ctx_fn())
```

Comment:
I'm not sure why this was failing without a unique `client_ctx` (in fact this may still fail, CI run isn't over yet)

Reply:
It's failing anyways. I don't get this!